addon/start-mirage.js

```diff
@@ -26,18 +26,29 @@ export default function startMirage(owner, { env, baseConfig, testConfig } = {})
     // These are set from `<app>/initializers/ember-cli-mirage`
     baseConfig = baseConfig || resolveRegistration(owner, 'mirage:base-config');
     testConfig = testConfig || resolveRegistration(owner, 'mirage:test-config');
+    makeServer = makeServer || resolveRegistration(owner, 'mirage:make-server');
   }
 
   let environment = env.environment;
   let mirageEnvironment = env['ember-cli-mirage'] || {};
   let discoverEmberDataModels = mirageEnvironment.discoverEmberDataModels;
   if (discoverEmberDataModels === undefined) { discoverEmberDataModels = true; }
   let modules = readModules(env.modulePrefix);
-  let options = Object.assign(modules, {environment, baseConfig, testConfig, discoverEmberDataModels});
+  let options = Object.assign(modules, {environment, routes: baseConfig, testConfig, discoverEmberDataModels});
   options.trackRequests = mirageEnvironment.trackRequests;
   options.inflector = { singularize, pluralize };
 
-  return new Server(options);
+  let server;
+  if (makeServer) {
+    server = makeServer(options);
+    if (typeof location !== 'undefined' && location.search.indexOf('mirageLogging') !== -1) {
```

Comment:
this guy supposed to be here?

Reply:
Yes, because when we call makeServer we are by passing the ember-cli-mirage Server class extend, and this code was in that Server class. Without it here the mirage logging check box will not work. I found this when i rebased my application to use a make server and I was running tests.

When the old way is deprecated and then removed, the Server class in ember-cli-mirage will also be removed.

Reply:
Ah the qunit toggle. Gotcha, thanks!